officialtech

```diff
@@ -0,0 +1 @@
+
```

Comment:
/*! @sentry/browser 6.4.1 (f9434ed) | https://github.com/getsentry/sentry-javascript */
var Sentry=function(t){var n=function(t,r){return(n=Object.setPrototypeOf||{__proto__:[]}instanceof Array&&function(t,n){t.__proto__=n}||function(t,n){for(var r in n)n.hasOwnProperty(r)&&(t[r]=n[r])})(t,r)};function r(t,r){function i(){this.constructor=t}n(t,r),t.prototype=null===r?Object.create(r):(i.prototype=r.prototype,new i)}var i,e,o,u,s,c,a=function(){return(a=Object.assign||function(t){for(var n,r=1,i=arguments.length;r<i;r++)for(var e in n=arguments[r])Object.prototype.hasOwnProperty.call(n,e)&&(t[e]=n[e]);return t}).apply(this,arguments)};function f(t){var n="function"==typeof Symbol&&t[Symbol.iterator],r=0;return n?n.call(t):{next:function(){return t&&r>=t.length&&(t=void 0),{value:t&&t[r++],done:!t}}}}function h(t,n){var r="function"==typeof Symbol&&t[Symbol.iterator];if(!r)return t;var i,e,o=r.call(t),u=[];try{for(;(void 0===n||n-- >0)&&!(i=o.next()).done;)u.push(i.value)}catch(t){e={error:t}}finally{try{i&&!i.done&&(r=o.return)&&r.call(o)}finally{if(e)throw e.error}}return u}function v(){for(var t=[],n=0;n<arguments.length;n++)t=t.concat(h(arguments[n]));return t}function l(t){switch(Object.prototype.toString.call(t)){case"[object Error]":case"[object Exception]":case"[object DOMException]":return!0;default:return x(t,Error)}}function d(t){return"[object ErrorEvent]"===Object.prototype.toString.call(t)}function p(t){return"[object DOMError]"===Object.prototype.toString.call(t)}function y(t){return"[object String]"===Object.prototype.toString.call(t)}function m(t){return null===t||"object"!=typeof t&&"function"!=typeof t}function b(t){return"[object Object]"===Object.prototype.toString.call(t)}function w(t){return"undefined"!=typeof Event&&x(t,Event)}function g(t){return"undefined"!=typeof Element&&x(t,Element)}function E(t){return Boolean(t&&t.then&&"function"==typeof t.then)}function x(t,n){try{return t instanceof n}catch(t){return!1}}function j(t){try{for(var n=t,r=[],i=0,e=0,o=" > ".length,u=void 0;n&&i++<5&&!("html"===(u=k(n))||i>1&&e+r.length*o+u.length>=80);)r.push(u),e+=u.length,n=n.parentNode;return r.reverse().join(" > ")}catch(t){return"<unknown>"}}function k(t){var n,r,i,e,o,u=t,s=[];if(!u||!u.tagName)return"";if(s.push(u.tagName.toLowerCase()),u.id&&s.push("#"+u.id),(n=u.className)&&y(n))for(r=n.split(/\s+/),o=0;o<r.length;o++)s.push("."+r[o]);var c=["type","name","title","alt"];for(o=0;o<c.length;o++)i=c[o],(e=u.getAttribute(i))&&s.push("["+i+'="'+e+'"]');return s.join("")}!function(t){t[t.None=0]="None",t[t.Error=1]="Error",t[t.Debug=2]="Debug",t[t.Verbose=3]="Verbose"}(i||(i={})),function(t){t.Ok="ok",t.Exited="exited",t.Crashed="crashed",t.Abnormal="abnormal"}(e||(e={})),function(t){t.Ok="ok",t.Errored="errored",t.Crashed="crashed"}(o||(o={})),(u=t.Severity||(t.Severity={})).Fatal="fatal",u.Error="error",u.Warning="warning",u.Log="log",u.Info="info",u.Debug="debug",u.Critical="critical",function(t){t.fromString=function(n){switch(n){case"debug":return t.Debug;case"info":return t.Info;case"warn":case"warning":return t.Warning;case"error":return t.Error;case"fatal":return t.Fatal;case"critical":return t.Critical;case"log":default:return t.Log}}}(t.Severity||(t.Severity={})),(s=t.Status||(t.Status={})).Unknown="unknown",s.Skipped="skipped",s.Success="success",s.RateLimit="rate_limit",s.Invalid="invalid",s.Failed="failed",function(t){t.fromHttpCode=function(n){return n>=200&&n<300?t.Success:429===n?t.RateLimit:n>=400&&n<500?t.Invalid:n>=500?t.Failed:t.Unknown}}(t.Status||(t.Status={})),function(t){t.Explicit="explicitly_set",t.Sampler="client_sampler",t.Rate="client_rate",t.Inheritance="inheritance"}(c||(c={}));var _=Object.setPrototypeOf||({__proto__:[]}instanceof Array?function(t,n){return t.__proto__=n,t}:function(t,n){for(var r in n)t.hasOwnProperty(r)||(t[r]=n[r]);return t});var S=function(t){function n(n){var r=this.constructor,i=t.call(this,n)||this;return i.message=n,i.name=r.prototype.constructor.name,_(i,r.prototype),i}return r(n,t),n}(Error),O=/^(?:(\w+):)\/\/(?:(\w+)(?::(\w+))?@)([\w.-]+)(?::(\d+))?\/(.+)/,D=function(){function t(t){"string"==typeof t?this.t(t):this.i(t),this.o()}return t.prototype.toString=function(t){void 0===t&&(t=!1);var n=this,r=n.host,i=n.path,e=n.pass,o=n.port,u=n.projectId;return n.protocol+"://"+n.publicKey+(t&&e?":"+e:"")+"@"+r+(o?":"+o:"")+"/"+(i?i+"/":i)+u},t.prototype.t=function(t){var n=O.exec(t);if(!n)throw new S("Invalid Dsn");var r=h(n.slice(1),6),i=r[0],e=r[1],o=r[2],u=void 0===o?"":o,s=r[3],c=r[4],a=void 0===c?"":c,f="",v=r[5],l=v.split("/");if(l.length>1&&(f=l.slice(0,-1).join("/"),v=l.pop()),v){var d=v.match(/^\d+/);d&&(v=d[0])}this.i({host:s,pass:u,path:f,projectId:v,port:a,protocol:i,publicKey:e})},t.prototype.i=function(t){"user"in t&&!("publicKey"in t)&&(t.publicKey=t.user),this.user=t.publicKey||"",this.protocol=t.protocol,this.publicKey=t.publicKey||"",this.pass=t.pass||"",this.host=t.host,this.port=t.port||"",this.path=t.path||"",this.projectId=t.projectId},t.prototype.o=function(){var t=this;if(["protocol","publicKey","host","projectId"].forEach(function(n){if(!t[n])throw new S("Invalid Dsn: "+n+" missing")}),!this.projectId.match(/^\d+$/))throw new S("Invalid Dsn: Invalid projectId "+this.projectId);if("http"!==this.protocol&&"https"!==this.protocol)throw new S("Invalid Dsn: Invalid protocol "+this.protocol);if(this.port&&isNaN(parseInt(this.port,10)))throw new S("Invalid Dsn: Invalid port "+this.port)},t}();function T(){return"[object process]"===Object.prototype.toString.call("undefined"!=typeof process?process:0)}function R(t,n){return void 0===n&&(n=0),"string"!=typeof t||0===n?t:t.length<=n?t:t.substr(0,n)+"..."}function I(t,n){if(!Array.isArray(t))return"";for(var r=[],i=0;i<t.length;i++){var e=t[i];try{r.push(String(e))}catch(t){r.push("[value cannot be serialized]")}}return r.join(n)}function N(t,n){return!!y(t)&&(r=n,"[object RegExp]"===Object.prototype.toString.call(r)?n.test(t):"string"==typeof n&&-1!==t.indexOf(n));var r}var M={};function C(){return T()?global:"undefined"!=typeof window?window:"undefined"!=typeof self?self:M}function A(){var t=C(),n=t.crypto||t.msCrypto;if(void 0!==n&&n.getRandomValues){var r=new Uint16Array(8);n.getRandomValues(r),r[3]=4095&r[3]|16384,r[4]=16383&r[4]|32768;var i=function(t){for(var n=t.toString(16);n.length<4;)n="0"+n;return n};return i(r[0])+i(r[1])+i(r[2])+i(r[3])+i(r[4])+i(r[5])+i(r[6])+i(r[7])}return"xxxxxxxxxxxx4xxxyxxxxxxxxxxxxxxx".replace(/[xy]/g,function(t){var n=16*Math.random()|0;return("x"===t?n:3&n|8).toString(16)})}function q(t){if(!t)return{};var n=t.match(/^(([^:\/?#]+):)?(\/\/([^\/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?$/);if(!n)return{};var r=n[6]||"",i=n[8]||"";return{host:n[4],path:n[5],protocol:n[2],relative:n[5]+r+i}}function L(t){if(t.message)return t.message;if(t.exception&&t.exception.values&&t.exception.values[0]){var n=t.exception.values[0];return n.type&&n.value?n.type+": "+n.value:n.type||n.value||t.event_id||"<unknown>"}return t.event_id||"<unknown>"}function U(t){var n=C();if(!("console"in n))return t();var r=n.console,i={};["debug","info","warn","error","log","assert"].forEach(function(t){t in n.console&&r[t].__sentry_original__&&(i[t]=r[t],r[t]=r[t].__sentry_original__)});var e=t();return Object.keys(i).forEach(function(t){r[t]=i[t]}),e}function H(t,n,r){t.exception=t.exception||{},t.exception.values=t.exception.values||[],t.exception.values[0]=t.exception.values[0]||{},t.exception.values[0].value=t.exception.values[0].value||n||"",t.exception.values[0].type=t.exception.values[0].type||r||"Error"}function F(t,n){void 0===n&&(n={});try{t.exception.values[0].mechanism=t.exception.values[0].mechanism||{},Object.keys(n).forEach(function(r){t.exception.values[0].mechanism[r]=n[r]})}catch(t){}}var P=6e4;var X=C(),B="Sentry Logger ",J=function(){function t(){this.u=!1}return t.prototype.disable=function(){this.u=!1},t.prototype.enable=function(){this.u=!0},t.prototype.log=function(){for(var t=[],n=0;n<arguments.length;n++)t[n]=arguments[n];this.u&&U(function(){X.console.log(B+"[Log]: "+t.join(" "))})},t.prototype.warn=function(){for(var t=[],n=0;n<arguments.length;n++)t[n]=arguments[n];this.u&&U(function(){X.console.warn(B+"[Warn]: "+t.join(" "))})},t.prototype.error=function(){for(var t=[],n=0;n<arguments.length;n++)t[n]=arguments[n];this.u&&U(function(){X.console.error(B+"[Error]: "+t.join(" "))})},t}();X.__SENTRY__=X.__SENTRY__||{};var W=X.__SENTRY__.logger||(X.__SENTRY__.logger=new J),$=function(){function t(){this.s="function"==typeof WeakSet,this.h=this.s?new WeakSet:[]}return t.prototype.memoize=function(t){if(this.s)return!!this.h.has(t)||(this.h.add(t),!1);for(var n=0;n<this.h.length;n++){if(this.h[n]===t)return!0}return this.h.push(t),!1},t.prototype.unmemoize=function(t){if(this.s)this.h.delete(t);else for(var n=0;n<this.h.length;n++)if(this.h[n]===t){this.h.splice(n,1);break}},t}(),G="<anonymous>";function z(t){try{return t&&"function"==typeof t&&t.name||G}catch(t){return G}}function K(t,n,r){if(n in t){var i=t[n],e=r(i);if("function"==typeof e)try{e.prototype=e.prototype||{},Object.defineProperties(e,{__sentry_original__:{enumerable:!1,value:i}})}catch(t){}t[n]=e}}function V(t){if(l(t)){var n=t,r={message:n.message,name:n.name,stack:n.stack};for(var i in n)Object.prototype.hasOwnProperty.call(n,i)&&(r[i]=n[i]);return r}if(w(t)){var e=t,o={};o.type=e.type;try{o.target=g(e.target)?j(e.target):Object.prototype.toString.call(e.target)}catch(t){o.target="<unknown>"}try{o.currentTarget=g(e.currentTarget)?j(e.currentTarget):Object.prototype.toString.call(e.currentTarget)}catch(t){o.currentTarget="<unknown>"}for(var i in"undefined"!=typeof CustomEvent&&x(t,CustomEvent)&&(o.detail=e.detail),e)Object.prototype.hasOwnProperty.call(e,i)&&(o[i]=e);return o}return t}function Q(t){return function(t){return~-encodeURI(t).split(/%..|./).length}(JSON.stringify(t))}function Y(t,n,r){void 0===n&&(n=3),void 0===r&&(r=102400);var i=nt(t,n);return Q(i)>r?Y(t,n-1,r):i}function Z(t,n){return"domain"===n&&t&&"object"==typeof t&&t.v?"[Domain]":"domainEmitter"===n?"[DomainEmitter]":"undefined"!=typeof global&&t===global?"[Global]":"undefined"!=typeof window&&t===window?"[Window]":"undefined"!=typeof document&&t===document?"[Document]":b(r=t)&&"nativeEvent"in r&&"preventDefault"in r&&"stopPropagation"in r?"[SyntheticEvent]":"number"==typeof t&&t!=t?"[NaN]":void 0===t?"[undefined]":"function"==typeof t?"[Function: "+z(t)+"]":"symbol"==typeof t?"["+String(t)+"]":"bigint"==typeof t?"[BigInt: "+String(t)+"]":t;var r}function tt(t,n,r,i){if(void 0===r&&(r=1/0),void 0===i&&(i=new $),0===r)return function(t){var n=Object.prototype.toString.call(t);if("string"==typeof t)return t;if("[object Object]"===n)return"[Object]";if("[object Array]"===n)return"[Array]";var r=Z(t);return m(r)?r:n}(n);if(null!=n&&"function"==typeof n.toJSON)return n.toJSON();var e=Z(n,t);if(m(e))return e;var o=V(n),u=Array.isArray(n)?[]:{};if(i.memoize(n))return"[Circular ~]";for(var s in o)Object.prototype.hasOwnProperty.call(o,s)&&(u[s]=tt(s,o[s],r-1,i));return i.unmemoize(n),u}function nt(t,n){try{return JSON.parse(JSON.stringify(t,function(t,r){return tt(t,r,n)}))}catch(t){return"**non-serializable**"}}function rt(t,n){void 0===n&&(n=40);var r=Object.keys(V(t));if(r.sort(),!r.length)return"[object has no keys]";if(r[0].length>=n)return R(r[0],n);for(var i=r.length;i>0;i--){var e=r.slice(0,i).join(", ");if(!(e.length>n))return i===r.length?e:R(e,n)}return""}function it(t){var n,r;if(b(t)){var i=t,e={};try{for(var o=f(Object.keys(i)),u=o.next();!u.done;u=o.next()){var s=u.value;void 0!==i[s]&&(e[s]=it(i[s]))}}catch(t){n={error:t}}finally{try{u&&!u.done&&(r=o.return)&&r.call(o)}finally{if(n)throw n.error}}return e}return Array.isArray(t)?t.map(it):t}function et(){if(!("fetch"in C()))return!1;try{return new Headers,new Request(""),new Response,!0}catch(t){return!1}}function ot(t){return t&&/^function fetch\(\)\s+\{\s+\[native code\]\s+\}$/.test(t.toString())}function ut(){if(!et())return!1;try{return new Request("_",{referrerPolicy:"origin"}),!0}catch(t){return!1}}var st,ct=C(),at={},ft={};function ht(t){if(!ft[t])switch(ft[t]=!0,t){case"console":!function(){if(!("console"in ct))return;["debug","info","warn","error","log","assert"].forEach(function(t){t in ct.console&&K(ct.console,t,function(n){return function(){for(var r=[],i=0;i<arguments.length;i++)r[i]=arguments[i];lt("console",{args:r,level:t}),n&&Function.prototype.apply.call(n,ct.console,r)}})})}();break;case"dom":!function(){if(!("document"in ct))return;var t=lt.bind(null,"dom"),n=wt(t,!0);ct.document.addEventListener("click",n,!1),ct.document.addEventListener("keypress",n,!1),["EventTarget","Node"].forEach(function(n){var r=ct[n]&&ct[n].prototype;r&&r.hasOwnProperty&&r.hasOwnProperty("addEventListener")&&(K(r,"addEventListener",function(n){return function(r,i,e){if("click"===r||"keypress"==r)try{var o=this.__sentry_instrumentation_handlers__=this.__sentry_instrumentation_handlers__||{},u=o[r]=o[r]||{refCount:0};if(!u.handler){var s=wt(t);u.handler=s,n.call(this,r,s,e)}u.refCount+=1}catch(t){}return n.call(this,r,i,e)}}),K(r,"removeEventListener",function(t){return function(n,r,i){if("click"===n||"keypress"==n)try{var e=this.__sentry_instrumentation_handlers__||{},o=e[n];o&&(o.refCount-=1,o.refCount<=0&&(t.call(this,n,o.handler,i),o.handler=void 0,delete e[n]),0===Object.keys(e).length&&delete this.__sentry_instrumentation_handlers__)}catch(t){}return t.call(this,n,r,i)}}))})}();break;case"xhr":!function(){if(!("XMLHttpRequest"in ct))return;var t=[],n=[],r=XMLHttpRequest.prototype;K(r,"open",function(r){return function(){for(var i=[],e=0;e<arguments.length;e++)i[e]=arguments[e];var o=this,u=i[1];o.__sentry_xhr__={method:y(i[0])?i[0].toUpperCase():i[0],url:i[1]},y(u)&&"POST"===o.__sentry_xhr__.method&&u.match(/sentry_key/)&&(o.__sentry_own_request__=!0);var s=function(){if(4===o.readyState){try{o.__sentry_xhr__&&(o.__sentry_xhr__.status_code=o.status)}catch(t){}try{var r=t.indexOf(o);if(-1!==r){t.splice(r);var e=n.splice(r)[0];o.__sentry_xhr__&&void 0!==e[0]&&(o.__sentry_xhr__.body=e[0])}}catch(t){}lt("xhr",{args:i,endTimestamp:Date.now(),startTimestamp:Date.now(),xhr:o})}};return"onreadystatechange"in o&&"function"==typeof o.onreadystatechange?K(o,"onreadystatechange",function(t){return function(){for(var n=[],r=0;r<arguments.length;r++)n[r]=arguments[r];return s(),t.apply(o,n)}}):o.addEventListener("readystatechange",s),r.apply(o,i)}}),K(r,"send",function(r){return function(){for(var i=[],e=0;e<arguments.length;e++)i[e]=arguments[e];return t.push(this),n.push(i),lt("xhr",{args:i,startTimestamp:Date.now(),xhr:this}),r.apply(this,i)}})}();break;case"fetch":!function(){if(!function(){if(!et())return!1;var t=C();if(ot(t.fetch))return!0;var n=!1,r=t.document;if(r&&"function"==typeof r.createElement)try{var i=r.createElement("iframe");i.hidden=!0,r.head.appendChild(i),i.contentWindow&&i.contentWindow.fetch&&(n=ot(i.contentWindow.fetch)),r.head.removeChild(i)}catch(t){W.warn("Could not create sandbox iframe for pure fetch check, bailing to window.fetch: ",t)}return n}())return;K(ct,"fetch",function(t){return function(){for(var n=[],r=0;r<arguments.length;r++)n[r]=arguments[r];var i={args:n,fetchData:{method:dt(n),url:pt(n)},startTimestamp:Date.now()};return lt("fetch",a({},i)),t.apply(ct,n).then(function(t){return lt("fetch",a(a({},i),{endTimestamp:Date.now(),response:t})),t},function(t){throw lt("fetch",a(a({},i),{endTimestamp:Date.now(),error:t})),t})}})}();break;case"history":!function(){if(t=C(),n=t.chrome,r=n&&n.app&&n.app.runtime,i="history"in t&&!!t.history.pushState&&!!t.history.replaceState,r||!i)return;var t,n,r,i;var e=ct.onpopstate;function o(t){return function(){for(var n=[],r=0;r<arguments.length;r++)n[r]=arguments[r];var i=n.length>2?n[2]:void 0;if(i){var e=st,o=String(i);st=o,lt("history",{from:e,to:o})}return t.apply(this,n)}}ct.onpopstate=function(){for(var t=[],n=0;n<arguments.length;n++)t[n]=arguments[n];var r=ct.location.href,i=st;if(st=r,lt("history",{from:i,to:r}),e)try{return e.apply(this,t)}catch(t){}},K(ct.history,"pushState",o),K(ct.history,"replaceState",o)}();break;case"error":gt=ct.onerror,ct.onerror=function(t,n,r,i,e){return lt("error",{column:i,error:e,line:r,msg:t,url:n}),!!gt&&gt.apply(this,arguments)};break;case"unhandledrejection":xt=ct.onunhandledrejection,ct.onunhandledrejection=function(t){return lt("unhandledrejection",t),!xt||xt.apply(this,arguments)};break;default:W.warn("unknown instrumentation type:",t)}}function vt(t){t&&"string"==typeof t.type&&"function"==typeof t.callback&&(at[t.type]=at[t.type]||[],at[t.type].push(t.callback),ht(t.type))}function lt(t,n){var r,i;if(t&&at[t])try{for(var e=f(at[t]||[]),o=e.next();!o.done;o=e.next()){var u=o.value;try{u(n)}catch(n){W.error("Error while triggering instrumentation handler.\nType: "+t+"\nName: "+z(u)+"\nError: "+n)}}}catch(t){r={error:t}}finally{try{o&&!o.done&&(i=e.return)&&i.call(e)}finally{if(r)throw r.error}}}function dt(t){return void 0===t&&(t=[]),"Request"in ct&&x(t[0],Request)&&t[0].method?String(t[0].method).toUpperCase():t[1]&&t[1].method?String(t[1].method).toUpperCase():"GET"}function pt(t){return void 0===t&&(t=[]),"string"==typeof t[0]?t[0]:"Request"in ct&&x(t[0],Request)?t[0].url:String(t[0])}var yt,mt,bt=1e3;function wt(t,n){return void 0===n&&(n=!1),function(r){if(r&&mt!==r&&!function(t){if("keypress"!==t.type)return!1;try{var n=t.target;if(!n||!n.tagName)return!0;if("INPUT"===n.tagName||"TEXTAREA"===n.tagName||n.isContentEditable)return!1}catch(t){}return!0}(r)){var i="keypress"===r.type?"input":r.type;void 0===yt?(t({event:r,name:i,global:n}),mt=r):function(t,n){if(!t)return!0;if(t.type!==n.type)return!0;try{if(t.target!==n.target)return!0}catch(t){}return!1}(mt,r)&&(t({event:r,name:i,global:n}),mt=r),clearTimeout(yt),yt=ct.setTimeout(function(){yt=void 0},bt)}}}var gt=null;var Et,xt=null;!function(t){t.PENDING="PENDING",t.RESOLVED="RESOLVED",t.REJECTED="REJECTED"}(Et||(Et={}));var jt=function(){function t(t){var n=this;this.l=Et.PENDING,this.p=[],this.m=function(t){n.g(Et.RESOLVED,t)},this.j=function(t){n.g(Et.REJECTED,t)},this.g=function(t,r){n.l===Et.PENDING&&(E(r)?r.then(n.m,n.j):(n.l=t,n.k=r,n._()))},this.S=function(t){n.p=n.p.concat(t),n._()},this._=function(){if(n.l!==Et.PENDING){var t=n.p.slice();n.p=[],t.forEach(function(t){t.done||(n.l===Et.RESOLVED&&t.onfulfilled&&t.onfulfilled(n.k),n.l===Et.REJECTED&&t.onrejected&&t.onrejected(n.k),t.done=!0)})}};try{t(this.m,this.j)}catch(t){this.j(t)}}return t.resolve=function(n){return new t(function(t){t(n)})},t.reject=function(n){return new t(function(t,r){r(n)})},t.all=function(n){return new t(function(r,i){if(Array.isArray(n))if(0!==n.length){var e=n.length,o=[];n.forEach(function(n,u){t.resolve(n).then(function(t){o[u]=t,0===(e-=1)&&r(o)}).then(null,i)})}else r([]);else i(new TypeError("Promise.all requires an array as input."))})},t.prototype.then=function(n,r){var i=this;return new t(function(t,e){i.S({done:!1,onfulfilled:function(r){if(n)try{return void t(n(r))}catch(t){return void e(t)}else t(r)},onrejected:function(n){if(r)try{return void t(r(n))}catch(t){return void e(t)}else e(n)}})})},t.prototype.catch=function(t){return this.then(function(t){return t},t)},t.prototype.finally=function(n){var r=this;return new t(function(t,i){var e,o;return r.then(function(t){o=!1,e=t,n&&n()},function(t){o=!0,e=t,n&&n()}).then(function(){o?i(e):t(e)})})},t.prototype.toString=function(){return"[object SyncPromise]"},t}(),kt=function(){function t(t){this.O=t,this.D=[]}return t.prototype.isReady=function(){return void 0===this.O||this.length()<this.O},t.prototype.add=function(t){var n=this;return this.isReady()?(-1===this.D.indexOf(t)&&this.D.push(t),t.then(function(){return n.remove(t)}).then(null,function(){return n.remove(t).then(null,function(){})}),t):jt.reject(new S("Not adding Promise due to buffer limit reached."))},t.prototype.remove=function(t){return this.D.splice(this.D.indexOf(t),1)[0]},t.prototype.length=functi